src/core/qgsfeaturepickermodelbase.cpp

```diff
@@ -309,14 +309,18 @@ void QgsFeaturePickerModelBase::updateCompleter()
     }
 
     // Remove all entries (except for extra entry if existent)
-    beginRemoveRows( QModelIndex(), firstRow, mEntries.size() - firstRow );
-    mEntries.remove( firstRow, mEntries.size() - firstRow );
+    if ( !mEntries.isEmpty() )
+    {
+      beginRemoveRows( QModelIndex(), firstRow, mEntries.size() - firstRow );
+      mEntries.remove( firstRow, mEntries.size() - firstRow );
+
+      // if we remove all rows before endRemoveRows, setExtraIdentifierValuesUnguarded will be called
+      // and a null value will be added to mEntries, so we block setExtraIdentifierValuesUnguarded call
 
-    // if we remove all rows before endRemoveRows, setExtraIdentifierValuesUnguarded will be called
-    // and a null value will be added to mEntries, so we block setExtraIdentifierValuesUnguarded call
+      mIsSettingExtraIdentifierValue = true;
+      endRemoveRows();
+    }
```

Comment:
Possibly needs this:

```suggestion
    }
    else
    {
        mIsSettingExtraIdentifierValue = true;
    }
```

Reply:
It's not just that. It looks like it relies on the side effects of calling beginRemoveRows even though there is actually no rows removed. The whole behavior seems fragile and too risky to modify without certainty it doesn't break everything. I'd rather close